CavalcanteLucas/todolist
# OL-48: User creation

**Link to task:** 
https://labcodes.atlassian.net/browse/OL-48

**How to test:**
- Create a user at `/register`;
- Check if success message is appearing with redirection to `/` after successful user creation;
- Check if button is disabled with any empty field or if double-check passwords are different;
- Check if fail message is appearing when trying to register the same user more than once.

**Description of your solution:**
With this PR one can register her/himself as a user in the system. This PR enables and tests an API for user registration, and adds some bootstrap-based styling. It also provides an interface with basic form validation. The user should not be able to submit the registration form if it is not properly filled. Successful registrations attempts should redirect to the index page and return a success message. Users with repeated usernames are not allowed; a failing message should appear if one tries to register with a username that is already registered.
